packages/braintree-integration/src/braintree-paypal/braintree-paypal-payment-strategy.ts

```diff
@@ -316,4 +484,35 @@ export default class BraintreePaypalPaymentStrategy implements PaymentStrategy {
 
         throw new PaymentMethodFailedError(error.message);
     }
+
+    private isProviderError(error: unknown): boolean {
+        if (isPaypalCommerceProviderError(error)) {
```

Comment:
We should not use paypal commerce related code inside brainteree strategy. Please create another typeguard for that.